fix(settings): overhaul settings layout

JPSchellenberg opened the pull request:
Hey,

before I start to integrate the library-management I have to overhaul the settings-section to make it more structured as we introduce more and more settings.

Let me know what you think.. I will add the translations (and the implementation of the account- & library-settings) later.

JPSchellenberg committed:
fix(settings): overhaul settings layout

JPSchellenberg committed:
test(lint): make ESLint happy

JPSchellenberg committed:
fix(settings): add locales for settings

JPSchellenberg committed:
fix(settings): remove deprecated components&hide unimplemented features

JPSchellenberg committed:
test(lint): make eslint happy

sr258 commented:
I like the way this is going! A few remarks:

- Users should not be able to withdraw consent to necessary data processing, as it makes no sense. The app is unusable then, so this setting should not exist.
- the view on the right-hand side (details-view / main area with all the settings) has too large horizontally: I can scroll it.
- the title of the "General" tab should also be "General" in the title of the actual settings. Currently it is "settings".
- It would be good if the currently selected tab was highlighted on the left.

![image](https://user-images.githubusercontent.com/15268740/113687633-d79dca00-96c8-11eb-8e5d-d39c4472c808.png)

JPSchellenberg committed:
Merge branch 'master' into fix/settings

JPSchellenberg committed:
fix(settings): scrollable main-area

JPSchellenberg committed:
fix(settings): highlight the current selected listitem

JPSchellenberg committed:
fix(settings): rework general listitem

JPSchellenberg commented:
I fixed some issues you mentioned and reworked the general section and divided it up to a general and updates section as I plan to improve the update process, which will require a separate updates section.
